controllers/discovery/initDiscoveryUserData.js

```diff
@@ -76,7 +78,7 @@ const InitDiscoveryUserData = async (req, res) => {
             ) as community_info,
             ( select count(name) > 1 from topics as tp
 							left join user_topics as utp on tp.topic_id = utp.topic_id
-							where utp.user_id = A.user_id and tp.topic_id in (:topicIds) LIMIT 2
+							where utp.user_id = A.user_id ${topicIdQuery} LIMIT 2
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The dynamic SQL query string `topicIdQuery` is correctly placed within the SQL query. It's important to note that if `topicIds` is null or empty, this condition will not be included in the SQL query, which is the intended behavior. However, as mentioned above, ensure that `topicIds` is properly sanitized before being used in the SQL query to prevent SQL injection attacks.
```diff
-    where utp.user_id = A.user_id ${topicIdQuery} LIMIT 2
+    where utp.user_id = A.user_id ${sanitize(topicIdQuery)} LIMIT 2
```
Please replace `sanitize()` with your actual sanitization function.


<!-- This is an auto-generated comment by OSS CodeRabbit -->